discovery-provider/src/tasks/cache_current_nodes.py

```diff
@@ -0,0 +1,31 @@
+import logging
+
+from src.tasks.celery_app import celery
+from src.utils.get_all_other_nodes import (
+    ALL_NODES_CACHE_KEY,
+    get_all_other_nodes,
+    get_node_endpoint,
+)
+from src.utils.prometheus_metric import save_duration_metric
+from src.utils.redis_cache import set_json_cached_key
+from src.utils.redis_connection import get_redis
+
+logger = logging.getLogger(__name__)
+
+
+# ####### CELERY TASKS ####### #
+@celery.task(name="cache_current_nodes")
+@save_duration_metric(metric_group="celery_task")
+def cache_current_nodes_task(self):
+    try:
+        nodes = get_all_other_nodes()[0]
+        current_node = get_node_endpoint()
```

Comment:
Not super necessary, but we add a redis lock to many celery tasks to prevent multiple from running at the same time. Just calling out.

https://github.com/AudiusProject/audius-protocol/blob/main/discovery-provider/src/tasks/index_oracles.py#L55